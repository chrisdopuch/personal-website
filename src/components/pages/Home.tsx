import Paper from '@material-ui/core/Paper';
import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React from 'react';
import me from '../../images/me.jpg';

const stylesDeclarations = (theme: Theme) =>
  createStyles({
    '@keyframes face-spin': {
      from: { transform: 'rotate(0deg)' },
      to: { transform: 'rotate(360deg)' },
    },
    face: {
      animation: 'face-spin infinite 20s linear',
      borderRadius: '50%',
      height: '40vmin',
      margin: '10px',
      maxHeight: '300px',
      maxWidth: '300px',
      width: 'auto',
    },
    root: {
      paddingBottom: theme.spacing.unit * 2,
      paddingTop: theme.spacing.unit * 2,
      ...theme.mixins.gutters(),
    },
  });

interface IHomeProps extends WithStyles<typeof stylesDeclarations> {}

class Home extends React.Component<IHomeProps> {
  public render() {
    const { classes } = this.props;
    return (
      <Paper className={classes.root} elevation={1}>
        <img src={me} className={classes.face} alt="my-face" />
        <Typography variant="body1">I wrote enough React to make my head spin!</Typography>
        <Typography variant="body1">More content to come soon...</Typography>
        <Typography variant="body1">
          <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
            Learn React
          </a>
        </Typography>
      </Paper>
    );
  }
}

export default withStyles(stylesDeclarations, { withTheme: true })(Home);
