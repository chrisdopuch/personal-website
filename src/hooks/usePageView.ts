import { useEffect } from 'react';
import GoogleAnalytics from 'react-ga';
import { RouteComponentProps } from 'react-router';

// a hook to track a page view in Google Analytics
export default function usePageView(location: RouteComponentProps['location'], options?: object) {
  const page = location.pathname + location.search;
  return useEffect(() => {
    GoogleAnalytics.set({
      page,
      ...options,
    });
    GoogleAnalytics.pageview(page);
  }, [page, options]); // only track the page once per mount
}
