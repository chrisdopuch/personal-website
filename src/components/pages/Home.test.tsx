import { shallow } from 'enzyme';
import { History, Location } from 'history';
import React from 'react';
import { match } from 'react-router';
import usePageView from '../hooks/usePageView';
import { Home } from './Home';

jest.mock('../hooks/usePageView', () => {
  return jest.fn();
});

const getDefaultProps = () => {
  return {
    classes: {
      '@keyframes face-spin': 'face-spin',
      divider: 'divider',
      face: 'face',
      root: 'root',
    },
    history: {} as History,
    location: {} as Location,
    match: {} as match,
  };
};

describe('Home', () => {
  it('should render with default props', () => {
    const wrapper = shallow(<Home {...getDefaultProps()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call the usePageView hook', () => {
    const props = {
      ...getDefaultProps(),
      location: {
        pathname: '/foo',
        search: '?bar=baz',
      } as Location,
    };
    shallow(<Home {...props} />);

    expect(usePageView).toBeCalledWith(props.location);
  });
});
