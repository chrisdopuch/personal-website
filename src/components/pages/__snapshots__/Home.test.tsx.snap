// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render with default props 1`] = `
<WithStyles(Paper)
  className="root"
  elevation={1}
>
  <WithStyles(Typography)
    variant="h2"
  >
    Home
  </WithStyles(Typography)>
  <WithStyles(Divider)
    className="divider"
  />
  <img
    alt="my-face"
    className="face"
    src="me.jpg"
  />
  <WithStyles(Typography)
    variant="body1"
  >
    I wrote enough React to make my head spin!
  </WithStyles(Typography)>
  <WithStyles(Typography)
    variant="body1"
  >
    More content to come soon...
  </WithStyles(Typography)>
  <WithStyles(Typography)
    variant="body1"
  >
    <a
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </WithStyles(Typography)>
</WithStyles(Paper)>
`;
